include/CLI/StringTools.hpp

```diff
@@ -163,39 +186,50 @@ inline std::string find_and_replace(std::string str, std::string from, std::stri
 }
 
 /// check if the flag definitions has possible false flags
-inline bool has_false_flags(const std::string &flags) { return (flags.find_first_of("{!") != std::string::npos); }
+inline bool has_default_flag_values(const std::string &flags) {
+    return (flags.find_first_of("{!") != std::string::npos);
+}
 
-inline void remove_false_flag_notation(std::string &flags) {
-    flags = detail::find_and_replace(flags, "{false}", std::string{});
-    flags = detail::find_and_replace(flags, "{true}", std::string{});
+inline void remove_default_flag_values(std::string &flags) {
+    size_t loc = flags.find_first_of('{');
+    while(loc != std::string::npos) {
+        auto finish = flags.find_first_of("},", loc + 1);
+        if((finish != std::string::npos) && (flags[finish] == '}')) {
+            flags.erase(flags.begin() + loc, flags.begin() + finish + 1);
+        }
+        loc = flags.find_first_of('{', loc + 1);
+    }
     flags.erase(std::remove(flags.begin(), flags.end(), '!'), flags.end());
 }
 
 /// Check if a string is a member of a list of strings and optionally ignore case or ignore underscores
-inline bool check_is_member(std::string name,
-                            const std::vector<std::string> names,
-                            bool ignore_case = false,
-                            bool ignore_underscore = false) {
+inline std::ptrdiff_t find_member(std::string name,
+                                  const std::vector<std::string> names,
+                                  bool ignore_case = false,
+                                  bool ignore_underscore = false) {
+    auto it = std::end(names);
     if(ignore_case) {
         if(ignore_underscore) {
```

Comment:
Would it make sense to to have a vector of transform functions, and allow a user to add any transforms they want; something like `opt->filter(CLI::ignore_underscore)->filter(CLI::ignore_case)`? Vector would be part of default option, etc. Just a thought for later.

Reply:
One thing I ran into was that translate functions should run before the actual validators so I had to make sure they were inserted at the front of the validator vector.    
So I think it is kind of an interesting usability discussion of whether check() should be allowed to modify or not.    On one had that differentiates check and translate but it disallows a CheckedTranslate since that would translate and make sure it was translated as a single unit.  You would need a translate and a separate check.